resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java

```diff
@@ -324,6 +326,11 @@ protected void initialize(ResteasyProviderFactoryImpl parent)
    {
       if (providerClasses == null && parent != null)

```

Comment:
`providerClasses ` will never be `null`, should we remove this if check to prevent confusion?  Or do you want to keep it just in case a later refactoring may potentially change things?

Reply:
I think you can remove the check

Reply:
Ok, I will push a new commit to remove these.